torch/distributions/utils.py

```diff
@@ -64,22 +48,20 @@ def broadcast_all(*values):
         ValueError: if any of the values is not a `numbers.Number` or
             `torch.*Tensor` instance
     """
-    values = list(values)
-    scalar_idxs = [i for i in range(len(values)) if isinstance(values[i], Number)]
-    tensor_idxs = [i for i in range(len(values)) if values[i].__class__.__name__ == 'Tensor']
-    if len(scalar_idxs) + len(tensor_idxs) != len(values):
+    if not all(map(lambda v: torch.is_tensor(v) or isinstance(v, Number), values)):
         raise ValueError('Input arguments must all be instances of numbers.Number or torch.tensor.')
-    if tensor_idxs:
-        broadcast_shape = _broadcast_shape([values[i].size() for i in tensor_idxs])
-        for idx in tensor_idxs:
-            values[idx] = values[idx].expand(broadcast_shape)
-        template = values[tensor_idxs[0]]
-        for idx in scalar_idxs:
-            values[idx] = template.new(template.size()).fill_(values[idx])
-    else:
-        for idx in scalar_idxs:
-            values[idx] = torch.tensor(float(values[idx]))
-    return values
+    if not all(map(torch.is_tensor, values)):
+        # promote numbers to tensors of dtype torch.get_default_dtype()
+        def default_promotion(v):
```

Comment:
nit: We should define this at file level to avoid the overhead of creating a new function on each invocation.
```py
def _default_promotion(v):
    return torch.tensor(v, dtype=torch.get_default_dtype())
```